mlflow/telemetry/utils.py

```diff
@@ -7,11 +7,56 @@
 from mlflow.environment_variables import MLFLOW_DISABLE_TELEMETRY
 
 
-# TODO: add check for CI  and mlflow dev versions,
-# and other scenarios where telemetry should be disabled
+def _is_ci_env() -> bool:
+    """
+    Check if the current environment is a CI environment.
+    If so, we should not track telemetry.
+    """
+    env_vars = {
+        "PYTEST_CURRENT_TEST",  # https://docs.pytest.org/en/stable/example/simple.html#pytest-current-test-environment-variable
```

Comment:
Does `PYTEST_CURRENT_TEST` defined really mean we're in CI?

Reply:
It's probably not CI, but it means pytest is used to run tests. Do you think this is too strict?

Reply:
Ok, let's rename the function to `is_ci_env_or_testing`